programs/uxd/src/instructions/credix_lp/exchange_liquidity_with_credix_lp_depository.rs

```diff
@@ -0,0 +1,334 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token;
+use anchor_spl::token::Mint;
+use anchor_spl::token::Token;
+use anchor_spl::token::TokenAccount;
+use anchor_spl::token::Transfer;
+
+use crate::error::UxdError;
+use crate::events::ExchangeLiquidityWithCredixLpDepositoryEvent;
+use crate::state::controller::Controller;
+use crate::state::credix_lp_depository::CredixLpDepository;
+use crate::state::identity_depository::IdentityDepository;
+use crate::utils::checked_add;
+use crate::utils::checked_as_u64;
+use crate::utils::checked_sub;
+use crate::utils::compute_decrease;
+use crate::utils::compute_increase;
+use crate::validate_is_program_frozen;
+use crate::CONTROLLER_NAMESPACE;
+use crate::CREDIX_LP_DEPOSITORY_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_COLLATERAL_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_NAMESPACE;
+
+#[derive(Accounts)]
+pub struct ExchangeLiquidityWithCredixLpDepository<'info> {
+    /// #1
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// #2
+    #[account(
+        mut,
+        seeds = [CONTROLLER_NAMESPACE],
+        bump = controller.load()?.bump,
+        constraint = controller.load()?.identity_depository == identity_depository.key() @UxdError::InvalidDepository,
+        constraint = controller.load()?.credix_lp_depository == credix_lp_depository.key() @UxdError::InvalidDepository,
+    )]
+    pub controller: AccountLoader<'info, Controller>,
+
+    /// #3
+    #[account(
+        mut,
+        seeds = [IDENTITY_DEPOSITORY_NAMESPACE],
+        bump = identity_depository.load()?.bump,
+    )]
+    pub identity_depository: AccountLoader<'info, IdentityDepository>,
+
+    /// #4
+    #[account(
+        mut,
+        seeds = [IDENTITY_DEPOSITORY_COLLATERAL_NAMESPACE],
+        token::authority = identity_depository,
+        token::mint = identity_depository.load()?.collateral_mint,
+        bump = identity_depository.load()?.collateral_vault_bump,
+    )]
+    pub identity_depository_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #5
+    #[account(
+        mut,
+        seeds = [CREDIX_LP_DEPOSITORY_NAMESPACE, credix_lp_depository.load()?.credix_global_market_state.key().as_ref(), credix_lp_depository.load()?.collateral_mint.as_ref()],
+        bump = credix_lp_depository.load()?.bump,
+        has_one = controller @UxdError::InvalidController,
+        has_one = collateral_mint @UxdError::InvalidCollateralMint,
+        constraint = credix_lp_depository.load()?.depository_shares == credix_lp_depository_shares.key() @UxdError::InvalidDepositoryShares,
+        constraint = credix_lp_depository.load()?.credix_shares_mint == credix_shares_mint.key() @UxdError::InvalidCredixSharesMint,
+    )]
+    pub credix_lp_depository: AccountLoader<'info, CredixLpDepository>,
+
+    /// #6
+    #[account(mut)]
+    pub credix_lp_depository_shares: Box<Account<'info, TokenAccount>>,
+
+    /// #7
+    pub user: Signer<'info>,
```

Comment:
nit: usually put all the Signer at the top of the ix

Reply:
Sounds good, done